src/renderer/js/redux/reducers/app.js

```diff
@@ -1,22 +1,60 @@
-import * as types from "constants/action_types";
-import * as modalTypes from "constants/modal_types";
+// @flow
 
+import * as types from "../../constants/action_types";
+import * as modalTypes from "../../constants/modal_types";
+
+// $FlowIssue: Flow cannot find election
 const { remote } = require("electron");
+
 const application = remote.app;
 const win = remote.BrowserWindow.getFocusedWindow();
 
 const reducers = {};
-const defaultState = {
+
+export type SnackBar = {
+  message: string,
+  linkText: string,
+  linkTarget: string,
+  isError: boolean,
+};
+export type AppState = {
+  isLoaded: boolean,
+  modal: ?string,
+  modalProps: mixed,
+  platform: string,
+  upgradeSkipped: boolean,
+  daemonVersionMatched: ?boolean,
+  daemonReady: boolean,
+  hasSignature: boolean,
+  badgeNumber: number,
+  volume: number,
+  downloadProgress: ?number,
+  upgradeDownloading: ?boolean,
+  upgradeDownloadComplete: ?boolean,
+  checkUpgradeTimer: ?number,
+  isUpgradeAvailable: ?boolean,
+  isUpgradeSkipped: ?boolean,
+  snackBar: ?SnackBar,
+};
+
+const defaultState: AppState = {
   isLoaded: false,
   modal: null,
   modalProps: {},
   platform: process.platform,
-  upgradeSkipped: sessionStorage.getItem("upgradeSkipped"),
+  upgradeSkipped: sessionStorage.getItem("upgradeSkipped") === "true",
   daemonVersionMatched: null,
   daemonReady: false,
   hasSignature: false,
   badgeNumber: 0,
-  volume: sessionStorage.getItem("volume") || 1,
+  volume: Number(sessionStorage.getItem("volume")) || 1,
+  downloadProgress: undefined,
```

Comment:
I am going to go through each of these one by one to determine whether or not I can add null, or some other initial value to them instead of undefined, however, there inevitably will be some fields that are initialized as undefined (given what we decided in terms of having no optional types + using undefined as a distinct value). Does this seem alright to you guys? @kauffj @seanyesmunt

Reply:
Yes, given that the existing behavior a property not explicitly set on `defaultState` is to default to `undefined`, it makes sense to set values to `undefined` rather than `null` to match existing behavior.

Reply:
Got it. Makes sense. I'm cool with this.